core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -348,15 +348,15 @@ class KafkaServer(
 
         /* start processing requests */
         dataPlaneRequestProcessor = new KafkaApis(socketServer.dataPlaneRequestChannel, replicaManager, adminManager, groupCoordinator, transactionCoordinator,
-          kafkaController, forwardingManager, zkClient, config.brokerId, config, metadataCache, metrics, authorizer, quotaManagers,
+          kafkaController, forwardingManager, zkClient, config.brokerId, config, replicaManager.configRepository, metadataCache, metrics, authorizer, quotaManagers,
```

Comment:
nit: Similarly here, it would be cleaner to construct the config repository separately in this class and pass it down to the components that need it including `ReplicaManager` and `KafkaApis`.

Reply:
@hachikuji After this change we have only 1 place left where we reach into `ReplicaManager` to grab its `ConfigRepository` -- when we create a `Partition` instance.  Do you think we should change that so that we explicitly provide the instance to the `Partition` constructor and the companion object's `apply()` method?  Aside form this last question I think we're all set here.

Reply:
It's just the `apply()` method that reaches in, so I'm going to add an explicit `ConfigRepository` parameter there.